Snowflake.Data/Core/Authenticator/IAuthenticator.cs

```diff
@@ -155,7 +155,7 @@ internal static IAuthenticator GetAuthenticator(SFSession session)
             else if (type.Equals(KeyPairAuthenticator.AUTH_NAME, StringComparison.InvariantCultureIgnoreCase))
             {
                 // Get private key path or private key from connection settings
-                if (!session.properties.TryGetValue(SFSessionProperty.PRIVATE_KEY_FILE, out var pkPath) &&
+                if ((!session.properties.TryGetValue(SFSessionProperty.PRIVATE_KEY_FILE, out var pkPath) || string.IsNullOrEmpty(pkPath)) &&
                     !session.properties.TryGetValue(SFSessionProperty.PRIVATE_KEY, out var pkContent))
```

Comment:
just wndering if here we should do something similar `|| string.IsNullOrEmpty(pkContent)`